src/deserialize.rs

```diff
@@ -0,0 +1,200 @@
+use crate::{Geometry, Wkt, WktFloat};
+use serde::de::{Deserializer, Error, Visitor};
+use std::{
+    default::Default,
+    fmt::{self, Debug},
+    marker::PhantomData,
+    str::FromStr,
+};
+
+struct WktVisitor<T> {
+    _marker: PhantomData<T>,
+}
+
+impl<T> Default for WktVisitor<T> {
+    fn default() -> Self {
+        WktVisitor {
+            _marker: PhantomData::default(),
+        }
+    }
+}
+
+impl<'de, T> Visitor<'de> for WktVisitor<T>
+where
+    T: FromStr + Default + Debug + WktFloat,
+{
+    type Value = Wkt<T>;
+    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(formatter, "a valid WKT format")
+    }
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        Wkt::from_str(s).map_err(|e| serde::de::Error::custom(e))
+    }
+}
+
+impl<'de, T> serde::Deserialize<'de> for Wkt<T>
+where
+    T: FromStr + Default + Debug + WktFloat,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(WktVisitor::default())
+    }
+}
+
+struct GeometryVisitor<T> {
+    _marker: PhantomData<T>,
+}
+
+impl<T> Default for GeometryVisitor<T> {
+    fn default() -> Self {
+        GeometryVisitor {
+            _marker: PhantomData::default(),
+        }
+    }
+}
+
+impl<'de, T> Visitor<'de> for GeometryVisitor<T>
+where
+    T: FromStr + Default + WktFloat,
+{
+    type Value = Geometry<T>;
+    fn expecting(&self, formatter: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(formatter, "a valid WKT format")
+    }
+    fn visit_str<E>(self, s: &str) -> Result<Self::Value, E>
+    where
+        E: Error,
+    {
+        let mut wkt = Wkt::from_str(s).map_err(|e| serde::de::Error::custom(e))?;
+        if wkt.items.len() == 1 {
+            Ok(wkt.items.remove(0))
+        } else {
+            Err(serde::de::Error::custom(
+                "WKT should have only 1 Geometry item",
+            ))
+        }
+    }
+}
+
+impl<'de, T> serde::Deserialize<'de> for Geometry<T>
+where
+    T: FromStr + Default + WktFloat,
+{
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_str(GeometryVisitor::default())
+    }
+}
+
+/// This is a helper function to convert directly from WKT format into a geo_types::Geometry.
+/// ```
+/// # extern crate wkt;
+/// # extern crate geo_types;
+/// # extern crate serde_json;
+/// use geo_types::Geometry;
+///
+/// #[derive(serde::Deserialize)]
+/// struct MyType {
+///     #[serde(deserialize_with = "wkt::deserialize_geometry")]
```

Comment:
oh right, there is a special dance to do when a doc-test requires a feature flag.

We have a similar example here in `proj` which requires the optional `geo-types` feature:

https://github.com/georust/proj/blob/master/src/lib.rs#L175

Reply:
Actually, forget that. This whole module is only included conditionally with the `serde` feature. I'm not immediately sure what to make of the CI failure. I can look at it further next week.

https://github.com/georust/wkt/runs/1891796241